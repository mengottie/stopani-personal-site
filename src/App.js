import React, { Component } from 'react';

import Navbar from './components/Navbar'
import SideMenu from './components/SideMenu';
import Backdrop from './components/Backdrop';
import PageWrapper from './components/PageWrapper';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      sideMenuIsOpen: false
     }
  }

  sideMenuToggle = () => {
    this.setState(prevState => {
      return {sideMenuIsOpen: !prevState.sideMenuIsOpen}
    })
  }
  
  render() {
    return (
      <>
        <Navbar toggleSideMenuHandler={this.sideMenuToggle} />
        { this.state.sideMenuIsOpen && 
          <>
            <Backdrop toggleSideMenuHandler={this.sideMenuToggle} /><SideMenu />
          </>}
        
        <PageWrapper />
      </>
    );
  }
}

export default App;
