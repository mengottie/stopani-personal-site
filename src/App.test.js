import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  const component = shallow(<App />);
  expect(component.find('a').text()).toBe('Learn React');
});
