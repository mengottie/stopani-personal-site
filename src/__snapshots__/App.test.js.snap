// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <Navbar
    toggleSideMenuHandler={[Function]}
  />,
  <div>
    Page wrapper
  </div>,
]
`;
